### ISSUE TITLE
[Design] Improved POS "Keypad" View

### ISSUE THREAD

**Issue by dstrukt** [btcpayserver/btcpayserver#3668]
I'm just posting this here for visibility, feedback, or suggestions as I'm still working through iterations. I'll be posting updates to this issue as I improve things. Not intended to be worked on just yet.

v0.1 Notes:
- Not shown in the prototype, but this view would be scrollable as it goes past the bottom row buttons .. but trying to make sure this quick prototype shows off all the current functionality, will work that into the next recording..
- The buttons would be fixed to the bottom, so would scale appropriately on larger height devices .. we could even center the keypad numbers as well.
- This prototype is missing the "Custom" tip input steps (to be added).
- Need a proper "<" icon (to be added)

![BTCPay_POS_Keypad](https://user-images.githubusercontent.com/6250771/164531184-2410dfd0-898c-434e-b347-0a52f44c9a4c.gif)

<img width="1388" alt="Screen Shot 2022-04-21 at 11 34 04 AM" src="https://user-images.githubusercontent.com/6250771/164531352-44c2e2bf-08a4-4d7d-b430-1735a0008b1b.png">

**Comment by Kukks** [comment]
This looks great! 

One thing I was considering at one point was to allow a side drawer to let you quickly pick from the predefined items if they are set.

**Comment by dstrukt** [comment]
Thanks! 

ooo that's an interesting concept, compact p.o.s .. I'm exploring the whole p.o.s flow in addition to the keypad view .. will add this to the list.

**Comment by leesalminen** [comment]
Just my 2c- our users would love to use the tip feature, but they need the tip amount stored somewhere on the invoice that's accessible to the Greenfield API. Currently, the tip amount is just tacked onto the total which makes reconciliation difficult enough that our users turned off tips. Let's get our restaurant servers some sats by making it easy for the business owner to implement!

**Comment by dstrukt** [comment]
Agree that this is incredibly important to see the full breakdown of price, from the invoice and details views, to the receipt .. will work towards making this visible everywhere we can. 

Will make a note to follow up with you if there's something we've missed!

**Comment by dennisreimann** [comment]
While reviewing #3991 I noticed that we can also improve the current state a bit with regards to values getting unset: Right now the assumption is, that the values get put in from top to bottom. Whenever one of the upper fields changes, it also resets the ones below it, which isn't optimal: When you change the amount, the values for discount and tip get unset, because they directly update the amount.

TL;DR We should distinguish between the amount field and the calculated value, which is the total sum in the end. That way we can change whichever field we want and recalculate the total sum, instead of updating the amount field and resetting everything else.

/cc @bolatovumar

**Comment by pavlenex** [comment]
@dstrukt Should we do this in 1.8.0? Seems we have a lot on the plate for 1.7.0 or do we already have a mock in mind that can be tackled by a dev?

**Comment by dennisreimann** [comment]
We can continue here once #4568 is merged.

**Comment by dstrukt** [comment]
![BTCPayPOS_Keypad_V0 2](https://user-images.githubusercontent.com/6250771/214916615-a28a642d-785b-448a-8402-f056a03d2952.gif)
